awx/ui_next/src/components/LaunchPrompt/LaunchPrompt.jsx

```diff
@@ -19,31 +19,50 @@ function LaunchPrompt({ config, resource, onLaunch, onCancel, i18n }) {
       component: <InventoryStep />,
     });
   }
-  // TODO: match old UI Logic:
-  // if (vm.promptDataClone.launchConf.ask_credential_on_launch ||
-  //     (_.has(vm, 'promptDataClone.prompts.credentials.passwords.vault') &&
-  //     vm.promptDataClone.prompts.credentials.passwords.vault.length > 0) ||
-  //     _.has(vm, 'promptDataClone.prompts.credentials.passwords.ssh_key_unlock') ||
-  //     _.has(vm, 'promptDataClone.prompts.credentials.passwords.become_password') ||
-  //     _.has(vm, 'promptDataClone.prompts.credentials.passwords.ssh_password')
-  // ) {
   if (config.ask_credential_on_launch) {
     initialValues.credentials = resource?.summary_fields?.credentials || [];
     steps.push({
-      name: i18n._(t`Credential`),
+      name: i18n._(t`Credentials`),
       component: <CredentialsStep />,
     });
   }
+
+  // TODO: Add Credential Passwords step
+
   if (
-    config.ask_scm_branch_on_launch ||
-    (config.ask_variables_on_launch && !config.ignore_ask_variables) ||
-    config.ask_tags_on_launch ||
-    config.ask_diff_mode_on_launch ||
-    config.ask_skip_tags_on_launch ||
     config.ask_job_type_on_launch ||
     config.ask_limit_on_launch ||
-    config.ask_verbosity_on_launch
+    config.ask_verbosity_on_launch ||
+    config.ask_tags_on_launch ||
+    config.ask_skip_tags_on_launch ||
+    config.ask_variables_on_launch ||
+    config.ask_scm_branch_on_launch ||
+    config.ask_diff_mode_on_launch
   ) {
+    if (config.ask_job_type_on_launch) {
+      initialValues.job_type = resource.job_type || '';
+    }
+    if (config.ask_limit_on_launch) {
+      initialValues.limit = resource.limit || '';
+    }
+    if (config.ask_verbosity_on_launch) {
+      initialValues.verbosity = resource.verbosity || 0;
+    }
+    if (config.ask_tags_on_launch) {
+      initialValues.job_tags = resource.job_tags || '';
+    }
+    if (config.ask_skip_tags_on_launch) {
+      initialValues.skip_tags = resource.skip_tags || '';
+    }
+    if (config.ask_variables_on_launch) {
+      initialValues.extra_vars = resource.extra_vars || '---';
+    }
+    if (config.ask_scm_branch_on_launch) {
+      initialValues.scm_branch = resource.scm_branch || '';
+    }
+    if (config.ask_diff_mode_on_launch) {
+      initialValues.diff_mode = resource.diff_mode || false;
+    }
```

Comment:
I'm wondering if these if's can be moved out of the enclosing if statement. Seems like the only thing that needs to be within this if statement is the `steps.push`?